### PR TITLE
Make some storage tests nondestructive

### PR DESCRIPTION
**Pull request by martinpitt**
Use the new RAM disk introduced in PR #13646

Also fix test failure on master that got introduced by unlucky timing (two parallel PRs)

### CODE REVIEW

**Comment by martinpitt** [comment]
Due to two PRs landing in parallel,  [testAddDisk](https://logs.cockpit-project.org/logs/pull-13650-20200304-111242-3bf8a918-fedora-31/log.html#79-2) is failing all the time in master now on Fedora 31. I fixed that now.

But the Machines cleanup is still imperfect. After it, when it tries to remove scsi_debug, the thing that keeps it busy is
```
/usr/bin/qemu-system-x86_64 -name guest=subVmTest1,debug-threads=on -S -object secret,id=masterKey0,format=raw,file=/var/lib/libvirt/qemu/domain-2-subVmTest1/master-key.aes -machine pc-i440fx-4.1,accel=tcg,usb=off,dump-guest-core=off -m 256 -overcommit mem-lock=off -smp 1,sockets=1,cores=1,threads=1 -uuid 9cce43b9-8a75-48a2-9957-2200e87a00f9 -no-user-config -nodefaults -chardev socket,id=charmonitor,fd=37,server,nowait -mon chardev=charmonitor,id=monitor,mode=control -rtc base=utc -no-shutdown -boot strict=on -device piix3-usb-uhci,id=usb,bus=pci.0,addr=0x1.0x2 -device virtio-scsi-pci,id=scsi0,bus=pci.0,addr=0x4 -device ahci,id=sata0,bus=pci.0,addr=0xa -device virtio-serial-pci,id=virtio-serial0,bus=pci.0,addr=0x5 -drive file=/var/lib/libvirt/images/subVmTest1-2.img,format=qcow2,if=none,id=drive-virtio-disk0 -device virtio-blk-pci,scsi=off,bus=pci.0,addr=0x6,drive=drive-virtio-disk0,id=virtio-disk0,bootindex=1,serial=SECOND -drive file=/mnt/vm_one/mydiskofpoolone_permanent,format=qcow2,if=none,id=drive-virtio-disk2,cache=writeback -device virtio-blk-pci,scsi=off,bus=pci.0,addr=0x8,drive=drive-virtio-disk2,id=virtio-disk2,write-cache=on -drive file=/mnt/vm_two/mydiskofpooltwo_permanent,format=qcow2,if=none,id=drive-virtio-disk3 -device virtio-blk-pci,scsi=off,bus=pci.0,addr=0x9,drive=drive-virtio-disk3,id=virtio-disk3 -drive file=/mnt/vm_one/sata_bus_disk,format=qcow2,if=none,id=drive-sata0-0-0 -device ide-hd,bus=sata0.0,drive=drive-sata0-0-0,id=sata0-0-0 -netdev tap,fd=39,id=hostnet0 -device rtl8139,netdev=hostnet0,id=net0,mac=52:54:00:28:7e:bd,bus=pci.0,addr=0x3,bootindex=2 -add-fd set=2,fd=41 -chardev file,id=charserial0,path=/dev/fdset/2,append=on -device isa-serial,chardev=charserial0,id=serial0 -chardev socket,id=charchannel0,fd=40,server,nowait -device virtserialport,bus=virtio-serial0.0,nr=1,chardev=charchannel0,id=channel0,name=org.qemu.guest_agent.0 -spice port=5900,addr=127.0.0.1,disable-ticketing,image-compression=off,seamless-migration=on -device VGA,id=video0,vgamem_mb=16,bus=pci.0,addr=0x2 -device virtio-balloon-pci,id=balloon0,bus=pci.0,addr=0x7 -sandbox on,obsolete=deny,elevateprivileges=deny,spawn=deny,resourcecontrol=deny -msg timestamp=on
```
I. e. it should try harder to kill all running domains. That VM eventually dies itself though, so the rmmod is eventually successful.

**Comment by martinpitt** [comment]
Oops, botched pyflakes and [this little problem](https://logs.cockpit-project.org/logs/pull-13650-20200304-135928-49472043-fedora-31/log.html), fixed.

**Comment by martinpitt** [comment]
The [debian-stable](https://logs.cockpit-project.org/logs/pull-13650-20200304-152416-5cfedeb7-debian-stable/log.html#166-2) failure is unrelated, I'll send a (proposed) fix.

**Comment by mvollmer** [comment]
Looks good to me!  I think there should be `rem_ram_disk` in StorageHelpers, though.

File: test/verify/check-storage-basic [hunk 1]
```diff
@@ -40,35 +43,34 @@ class TestStorage(StorageCase):
...
 
         b.go("#/")
         b.wait_visible('#storage')
-        b.wait_in_text("#drives", "MYSERIAL")
-        m.execute('udevadm settle')
-        m.rem_disk(disk1)
-        b.wait_not_in_text("#drives", "MYSERIAL")
+        b.wait_in_text("#drives", "scsi_debug")
+        # forcefully remove the drive
+        m.execute('echo 1 > /sys/block/%s/device/delete' % os.path.basename(disk))
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply test/verify/check-storage-basic]: Done.

File: test/verify/check-storage-partitions [hunk 2]
```diff
@@ -22,6 +22,7 @@ from storagelib import *
 from testlib import *
 
 
+@nondestructive
 class TestStorage(StorageCase):
 
     @nondestructive
```
[the remark this answers is not in the document]
Comment (martinpitt) [reply test/verify/check-storage-partitions]: Yes, absolutely, done.

**Comment by martinpitt** [comment]
@mvollmer: Thanks for the review! PTAL?

**Comment by martinpitt** [comment]
Oops, fixed pyflakes error.

**Pull request merged**